### PR TITLE
Add some user forwards

### LINKED ISSUE #287: issues with pad_constant
recently merged  #283 introduces a few issues. According to its own docstring
```
"""
    pad_constant(x, pad::Tuple, val = 0; [dims])
    pad_constant(x, pad::Int, val = 0; [dims])

Pad the array `x` with the constant value `val`.
`pad` can a tuple of integers `(l1, r1, ..., ln, rn)`
of some length `2n` that specifies the left and right padding size
for each of the dimensions in `dims`. If `dims` is not given, 
it defaults to the first `n` dimensions.
For integer `pad` input instead, it is applied on both sides
on every dimension in `dims`. In this case, `dims` 
defaults to the first `ndims(x)-2` dimension 
(i.e. excludes the channel and batch dimension).
"""
```
which is also consistent with other padding functions do, we have the following inconsistencies:
```julia
julia> r = reshape(1:8, 2, 2, 2)
2×2×2 reshape(::UnitRange{Int64}, 2, 2, 2) with eltype Int64:
[:, :, 1] =
 1  3
 2  4

[:, :, 2] =
 5  7
 6  8

# INCONSISTENT WITH OWN DOCSTRING AND OTHER PAD METHODS + ERROR NOT INFORMATIVE
julia> pad_constant(r, (1, 1, 1, 1)) # 
ERROR: ArgumentError: Could not parse padding (1, 1, 1, 1) and dims (1, 2, 3)
Stacktrace:
 [1] #251
   @ ~/.julia/dev/NNlib/src/padding.jl:83 [inlined]
 [2] ntuple
   @ ./ntuple.jl:19 [inlined]
 [3] gen_pad(pad::NTuple{4, Int64}, dims::Tuple{Int64, Int64, Int64}, N::Int64)
   @ NNlib ~/.julia/dev/NNlib/src/padding.jl:74
 [4] gen_pad(pad::NTuple{4, Int64}, dims::Colon, N::Int64)
   @ NNlib ~/.julia/dev/NNlib/src/padding.jl:72
 [5] pad_constant(x::Base.ReshapedArray{Int64, 3, UnitRange{Int64}, Tuple{}}, pad::NTuple{4, Int64}, val::Int64; dims::Function)
   @ NNlib ~/.julia/dev/NNlib/src/padding.jl:49
 [6] pad_constant
   @ ~/.julia/dev/NNlib/src/padding.jl:49 [inlined]
 [7] top-level scope
   @ REPL[22]:1

# CANNOT TAKE INT DIMS, pad_constant(r, (1, 2), dims=(1,)) WORKS FINE       
julia> pad_constant(r, (1, 2), dims=1)
ERROR: MethodError: no method matching gen_pad(::Tuple{Int64, Int64}, ::Int64, ::Int64)
Closest candidates are:
  gen_pad(::Int64, ::Any, ::Any) at /home/carlo/.julia/dev/NNlib/src/padding.jl:70
  gen_pad(::Any, ::Colon, ::Any) at /home/carlo/.julia/dev/NNlib/src/padding.jl:72
  gen_pad(::Tuple{Vararg{Int64, L}}, ::Tuple{Vararg{Int64, D}}, ::Any) where {L, D} at /home/carlo/.julia/dev/NNlib/src/padding.jl:73
  ...
Stacktrace:
 [1] pad_constant(x::Base.ReshapedArray{Int64, 3, UnitRange{Int64}, Tuple{}}, pad::Tuple{Int64, Int64}, val::Int64; dims::Int64)
   @ NNlib ~/.julia/dev/NNlib/src/padding.jl:49
 [2] top-level scope
   @ REPL[26]:1

# CANNOT TAKE RANGE DIMS, pad_constant(r, (1, 2), dims=(1,2)) WORKS FINE       
julia> pad_constant(r, (1, 1, 1, 1), dims=1:2)
ERROR: MethodError: no method matching gen_pad(::NTuple{4, Int64}, ::UnitRange{Int64}, ::Int64)
Closest candidates are:
  gen_pad(::Int64, ::Any, ::Any) at /home/carlo/.julia/dev/NNlib/src/padding.jl:70
  gen_pad(::Any, ::Colon, ::Any) at /home/carlo/.julia/dev/NNlib/src/padding.jl:72
  gen_pad(::Tuple{Vararg{Int64, L}}, ::Tuple{Vararg{Int64, D}}, ::Any) where {L, D} at /home/carlo/.julia/dev/NNlib/src/padding.jl:73
  ...
Stacktrace:
 [1] pad_constant(x::Base.ReshapedArray{Int64, 3, UnitRange{Int64}, Tuple{}}, pad::NTuple{4, Int64}, val::Int64; dims::UnitRange{Int64})
   @ NNlib ~/.julia/dev/NNlib/src/padding.jl:49
 [2] top-level scope
   @ REPL[27]:1

# PADS ALL DIMENSIONS INSTEAD OF JUST FIRST ACCORDING TO DOCSTRING AND OTHER PAD METHODS
julia> pad_constant(r, 1) |> size
(4, 4, 4)
```
None of these issues were there before #283 . Should we revert it?
@DhairyaLGandhi

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
Closes #287

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Can you add tests and change the other padding functions accordingly?

**Comment by DhairyaLGandhi** [comment]
I first need to fix the release.

File: src/padding.jl [hunk 1]
```diff
@@ -42,12 +41,67 @@ julia> pad_constant(r, (1, 2, 3, 4), 8)
...
  8  8  8  8  8  8  8  8  8
  8  8  8  8  8  8  8  8  8
+
+julia> pad_constant(r, 1, 8)
+5×9 Matrix{Int64}:
+ 8  8  8  8
+ 8  1  3  8
+ 8  2  4  8
+ 8  8  8  8
 ```
```
Comment (CarloLucibello) [inline src/padding.jl:51]: these ticks are in the wrong position. Also could reduce the number of examples, if the explanation of the behavior is clear

File: src/padding.jl [hunk 2]
```diff
@@ -11,20 +11,19 @@ pad_zeros(x::AbstractArray, pad; dims = :) =
...
 
 Pad the array `x` with the constant value `val`.
 
-`pad` can a tuple of integers `(l1, r1, ..., ln, rn)`
-of some length `2n` that specifies the left and right padding size
-for each of the dimensions in `dims`. If `dims` is not given, 
-it defaults to the first `n` dimensions.
+`pad` can take a tuple of length equal or double the length of
+`dims`. In case of equal length, it applies uniform padding on both sides
+and for length `2*length(dims)` the padding is applied assymetrically.
```
Comment (CarloLucibello) [inline src/padding.jl:21]:
```suggestion
`pad` can be a tuple of integers `(l1, r1, ..., ln, rn)`
of some length `2n` that specifies the left and right padding size
for each of the dimensions in `dims`. 
If supplied with a tuple of length `n`  instead, it applies symmetric padding.
If `dims` is not given, it defaults to the number of dimensions in the array.
```

File: src/padding.jl [hunk 3]
```diff
@@ -11,20 +11,19 @@ pad_zeros(x::AbstractArray, pad; dims = :) =
...
+
+By default, padding is applied to every dimension.
```
Comment (CarloLucibello) [inline src/padding.jl:23]:
```suggestion
```

**Pull request merged**